components/survey/personal-information.tsx

```diff
@@ -1,15 +1,237 @@
-import { View, Text } from "react-native";
-import React from "react";
-import Header from "../common/Header";
+import React, { useMemo, useState, useEffect } from "react";
+import {
+  View,
+  Text,
+  TextInput,
+  Pressable,
+  Platform,
+  ScrollView,
+  KeyboardAvoidingView,
+} from "react-native";
+import Header from "@/components/common/Header";
+import { RadioButtonProps } from "react-native-radio-buttons-group";
+import DateTimePicker, {
+  DateTimePickerEvent,
+} from "@react-native-community/datetimepicker";
+import * as Location from "expo-location";
 
 const PersonalInformation = () => {
+  const [selectedId, setSelectedId] = useState<string | undefined>();
+  const [birthday, setBirthday] = useState("");
+  const [date, setDate] = useState<Date>(new Date());
+  const [show, setShow] = useState(false);
+  const [location, setLocation] = useState<Location.LocationObject | null>(
+    null
+  );
+  const [errorMsg, setErrorMsg] = useState<string | null>(null);
+  const [address, setAddress] = useState<string | null>(null);
+
+  const gender: RadioButtonProps[] = useMemo(
+    () => [
+      { id: "1", label: "Man" },
+      { id: "2", label: "Woman" },
+      { id: "3", label: "Non-Binary" },
+      { id: "4", label: "Prefer not to say" },
+    ],
+    []
+  );
+
+  const toggleShow = () => {
+    setShow(!show);
+  };
+
+  const formatDate = (date: Date) => {
+    const options: Intl.DateTimeFormatOptions = {
+      month: "long",
+      day: "2-digit",
+      year: "numeric",
+    };
+    const formattedDate = new Intl.DateTimeFormat("en-US", options).format(
+      date
+    );
+
+    const [month, day, year] = formattedDate.replace(",", "").split(" ");
+
+    return `${month} | ${day} | ${year}`;
+  };
+
+  const onChange = (
+    _event: DateTimePickerEvent,
+    selectedDate: Date | undefined
+  ) => {
+    if (selectedDate) {
+      setDate(selectedDate);
+      setBirthday(formatDate(selectedDate));
+    }
+    setShow(false);
+  };
+
+  const requestLocationPermission = async () => {
```

Comment:
I think pwede ta mn ni ma refactor since laba siya sa file and damo ang conditionals